library/src/scala/App.scala

```diff
@@ -62,7 +62,7 @@ import scala.collection.mutable.ListBuffer
  *  }
  *  }}}
  */
-@nowarn("cat=deprecation")
+@deprecated(message = "Support for trait App is deprecated in Scala 3. Please refer to https://docs.scala-lang.org/scala3/book/methods-main-methods.html.", since = "3.0.0")
```

Comment:
```suggestion
@deprecated(message = "Support for trait App is deprecated in Scala 3. Please refer to https://docs.scala-lang.org/scala3/book/methods-main-methods.html.", since = "3.8.0")
```
Clearly, `App` *wasn't* deprecated before 3.8.0.

Reply:
I hesitated a bit on either putting `3.8.0` or `3.0.0` given the status of `App`. I don't have any strong opinions about it.